src/web.ts

```diff
@@ -0,0 +1,532 @@
+    display: none;
+  }
+}
+
+.ad4mConnect__backdrop {
+	position: absolute;
+	top: 0;
+	left: 0;
+	height: 100vh;
+	width: 100vw;
+	background-color: rgba(0, 0, 0, 0.4);
+}
+
+.ad4mConnect__dailog__header__logo {
+	height: 30px;
+	width: 30px;
+}
+
+.ad4mConnect__dailog__title {
+	font-weight: bold;
+	font-size: 24px;
+	margin-left: 20px;
+  color: #fff;
+}
+
+.ad4mConnect__dailog__header {
+	display: flex;
+	align-items: center;
+}
+
+.ad4mConnect__dailog__subtitle {
+	margin: 18px 0;
+  color: #fff;
+}
+
+.ad4mConnect__dailog__caps {
+	align-self: flex-start;
+  color: #fff;
+  width: 100%;
+}
+
+.ad4mConnect__dailog__input {
+  color: #fff;
+  font-size: 18px;
+  font-weight: bold;
+  margin: 24px 0;
+  align-self: flex-start;
+}
+
+.ad4mConnect__dailog__input input {
+  font-size: 16px;
+  background: transparent;
+  color: #fff;
+  border: 1px solid #fff;
+  border-radius: 4px;
+  padding: 12px;
+  margin-left: 12px;
+  max-width: 400px;
+}
+
+.ad4mConnect__dailog__connection {
+	display: flex;
+	align-items: center;
+	margin: 24px 0;
+}
+
+.ad4mConnect__dailog__btn {
+	padding: 12px 24px;
+	border-radius: 4px;
+	border: 1px solid #fff;
+}
+
+.ad4mConnect__dailog__btns {
+	display: flex;
+	align-self: flex-end;
+}
+
+.ad4mConnect__locked {
+	position: fixed;
+	top: 0;
+	left: 0;
+	background: linear-gradient(90deg, rgba(2, 0, 36, 1) 0%, rgba(38,3,23,1) 41%, rgba(51,4,31,1) 100%);
+	height: 100vh;
+	width: 100vw;
+	padding: 36px;
+	display: flex;
+	align-items: center;
+	flex-direction: column;
+  font-family: 'Comfortaa', cursive;
+}
+
+.ad4mConnect__dailog__options {
+	width: 100%;
+}
+
+.ad4mConnect__dailog__option {
+	background-color: #eee;
+  width: calc(100% - 48px);
+	display: flex;
+	align-items: center;
+	padding: 24px 24px;
+	margin-top: 12px;
+	border-radius: 4px;
+	font-size: 18px;
+  cursor: pointer;
+}
+
+.ad4mConnect__dailog__option__text {
+  font-weight: bold;
+}
+
+.lds-ring {
+  display: inline-block;
+  position: relative;
+  width: 80px;
+  height: 80px;
+  margin-top: 24px;
+}
+.lds-ring div {
+  box-sizing: border-box;
+  display: block;
+  position: absolute;
+  width: 64px;
+  height: 64px;
+  margin: 8px;
+  border: 8px solid #fff;
+  border-radius: 50%;
+  animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
+  border-color: #fff transparent transparent transparent;
+}
+.lds-ring div:nth-child(1) {
+  animation-delay: -0.45s;
+}
+.lds-ring div:nth-child(2) {
+  animation-delay: -0.3s;
+}
+.lds-ring div:nth-child(3) {
+  animation-delay: -0.15s;
+}
+@keyframes lds-ring {
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
+
+`
+
+@customElement("ad4m-connect")
+export default class Ad4mConnect extends LitElement {
+  static styles = [styles];
+
+  @state()
+  private _client = null;
+
+  @state()
+  private _state = null;
+
+  @state()
+  private _code = null;
+
+  @state()
+  private _url = null;
+
+  @property({ type: String, reflect: true })
+  appname = null
```

Comment:
Will it be better if we use `app` attributes to include `appname`, `appdesc`, `appdomain`?
Also, the naming of `appname` should be snake case.

Reply:
attributes name cannot be be snake case that's why this also lit element doesn't do camelcase either https://github.com/lit/lit-element/issues/29